Sources/Build/misc.swift

```diff
@@ -32,6 +35,83 @@ extension CModule {
     }
 }
 
+extension ClangModule {
+    
+    public enum ModuleMapError: ErrorProtocol {
+        case UnsupportedIncludeLayoutForModule(String)
+    }
+    
+    public func generateModuleMap() throws {
+        
+        //Return if module map is already present
+        guard !moduleMapPath.isFile else {
```

Comment:
Yeah there are two problems currently...
* module maps should probably be generated in build dir or somewhere
* incremental map generation